pulser/sequence.py

```diff
@@ -12,32 +12,50 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from __future__ import annotations
+
 from collections import namedtuple
-from collections.abc import Iterable
+from collections.abc import Callable, Generator, Iterable, Set
 import copy
 from functools import wraps
 from itertools import chain
 import json
+from typing import Any, cast, NamedTuple, Optional, Tuple, Union
+from typing_extensions import Literal
```

Comment:
I understand you must be saturated at this point, but I found a solution where we can use `get_args` as before. If you want, I can also merge this and do it myself. What do you prefer?

Reply:
Let me know and I'll try...

Reply:
Alright. In fact, you can import `get_args` from `typing_extensions`, so your first solution would work. However, I would rather we import from `typing_extensions` when necessary (i.e. when the python version being used is 3.7.). So, here's what I would suggest:
```suggestion
import platform
if platform.python_version()[:3] == '3.7':
    try:
        from typing_extensions import Literal, get_args
    except ImportError:
        raise ImportError("Using pulser with Python version 3.7 requires the `typing_extensions` module. Install it by running `pip install typing-extensions`.")
else:
    from typing import Literal, get_args
```
(you'll have to fix the style, it's hard to do it here)

Then you should be able to use `get_args` as before and avoid duplicating `valid_protocols`.

Reply:
Also, perhaps using `sys.version_info` might be preferable over `platform.python_version()`, which I suggested above. Just don't use `sys.version`